lib/travis/api/v3/services/lead/create.rb

```diff
@@ -0,0 +1,41 @@
+require 'uri'
+require 'closeio'
+
+module Travis::API::V3
+  class Services::Lead::Create < Service
+    result_type :lead
+    params :name, :email, :team_size, :phone, :message, :utm_source
+
+    def run!
+      # Validation
+      raise WrongParams, 'missing name' unless params['name'] && params['name'].length > 0
+      raise WrongParams, 'invalid email' unless params['email'] && params['email'].length > 0 && params['email'].match(URI::MailTo::EMAIL_REGEXP).present?
+      raise WrongParams, 'missing message' unless params['message'] && params['message'].length > 0
+
+      # Prep data for request
+      name, email, team_size, phone, message, utm_source = params.values_at('name', 'email', 'team_size', 'phone', 'message', 'utm_source')
+      phones = []
+      phones.push({ type: "office", phone: phone }) unless phone.nil?
+
+      lead_data = {
+        name: name,
+        'custom.team_size': team_size,
+        'custom.utm_source': utm_source || 'Travis API',
```

Comment:
Sorry for missing it earlier, but custom fields in Close.io must be referenced by their `id`s, not names. Please take a look at the implementation in [Enterprise landing page](https://github.com/travis-pro/enterprise-landing-page/blob/master/app/services/closeio_lead.rb#L39-L61). Otherwise Close.io API will reject the request.

Also `team_size` must be a number, Close.io API rejects even numerical strings, so `team_size.to_i` might be necessary.

These are the issues that I faced while working on enterprise page update. Though, I see the leads in Close.io that were created using your current API implementation, so I assume, it works 😄 In that case please consider these comments as completely optional! No need to "fix" things that are working properly ;)

Reply:
Ahh nice catch! I think if this is the more proper way to do it, we might as well make the change now. So yeah, updates implemented :smile: